// plugin-welcome/src/services/api/apis/DefaultApi.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */


import * as runtime from '../runtime';
import {
    EntDentist,
    EntDentistFromJSON,
    EntDentistToJSON,
    EntEmployee,
    EntEmployeeFromJSON,
    EntEmployeeToJSON,
    EntPatient,
    EntPatientFromJSON,
    EntPatientToJSON,
    EntPillorder,
    EntPillorderFromJSON,
    EntPillorderToJSON,
    EntPillorderItem,
    EntPillorderItemFromJSON,
    EntPillorderItemToJSON,
} from '../models';

export interface CreateDentistRequest {
    dentist: EntDentist;
}

export interface CreateEmployeeRequest {
    employee: EntEmployee;
}

export interface CreatePatientRequest {
    patient: EntPatient;
}

export interface CreatePillorderRequest {
    pillorder: EntPillorder;
}

export interface CreatePillorderitemRequest {
    pillorderitem: EntPillorderItem;
}

export interface GetDentistRequest {
    id: number;
}

export interface GetEmployeeRequest {
    id: number;
}

export interface GetPatientRequest {
    id: number;
}

export interface GetPillorderRequest {
    id: number;
}

export interface GetPillorderitemRequest {
    id: number;
}

export interface ListDentistRequest {
    limit?: number;
    offset?: number;
}

export interface ListEmployeeRequest {
    limit?: number;
    offset?: number;
}

export interface ListPatientRequest {
    limit?: number;
    offset?: number;
}

export interface ListPillorderRequest {
    limit?: number;
    offset?: number;
}

export interface ListPillorderitemRequest {
    limit?: number;
    offset?: number;
}

/**
 * 
 */
export class DefaultApi extends runtime.BaseAPI {

    /**
     * Create dentist
     * Create dentist
     */
    async createDentistRaw(requestParameters: CreateDentistRequest): Promise<runtime.ApiResponse<EntDentist>> {
        if (requestParameters.dentist === null || requestParameters.dentist === undefined) {
            throw new runtime.RequiredError('dentist','Required parameter requestParameters.dentist was null or undefined when calling createDentist.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        const response = await this.request({
            path: `/dentists`,
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: EntDentistToJSON(requestParameters.dentist),
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntDentistFromJSON(jsonValue));
    }

    /**
     * Create dentist
     * Create dentist
     */
    async createDentist(requestParameters: CreateDentistRequest): Promise<EntDentist> {
        const response = await this.createDentistRaw(requestParameters);
        return await response.value();
    }

    /**
     * Create employee
     * Create employee
     */
    async createEmployeeRaw(requestParameters: CreateEmployeeRequest): Promise<runtime.ApiResponse<EntEmployee>> {
        if (requestParameters.employee === null || requestParameters.employee === undefined) {
            throw new runtime.RequiredError('employee','Required parameter requestParameters.employee was null or undefined when calling createEmployee.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        const response = await this.request({
            path: `/employees`,
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: EntEmployeeToJSON(requestParameters.employee),
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntEmployeeFromJSON(jsonValue));
    }

    /**
     * Create employee
     * Create employee
     */
    async createEmployee(requestParameters: CreateEmployeeRequest): Promise<EntEmployee> {
        const response = await this.createEmployeeRaw(requestParameters);
        return await response.value();
    }

    /**
     * Create patient
     * Create patient
     */
    async createPatientRaw(requestParameters: CreatePatientRequest): Promise<runtime.ApiResponse<EntPatient>> {
        if (requestParameters.patient === null || requestParameters.patient === undefined) {
            throw new runtime.RequiredError('patient','Required parameter requestParameters.patient was null or undefined when calling createPatient.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        const response = await this.request({
            path: `/patients`,
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: EntPatientToJSON(requestParameters.patient),
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntPatientFromJSON(jsonValue));
    }

    /**
     * Create patient
     * Create patient
     */
    async createPatient(requestParameters: CreatePatientRequest): Promise<EntPatient> {
        const response = await this.createPatientRaw(requestParameters);
        return await response.value();
    }

    /**
     * Create pillorder
     * Create pillorder
     */
    async createPillorderRaw(requestParameters: CreatePillorderRequest): Promise<runtime.ApiResponse<EntPillorder>> {
        if (requestParameters.pillorder === null || requestParameters.pillorder === undefined) {
            throw new runtime.RequiredError('pillorder','Required parameter requestParameters.pillorder was null or undefined when calling createPillorder.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        const response = await this.request({
            path: `/pillorders`,
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: EntPillorderToJSON(requestParameters.pillorder),
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntPillorderFromJSON(jsonValue));
    }

    /**
     * Create pillorder
     * Create pillorder
     */
    async createPillorder(requestParameters: CreatePillorderRequest): Promise<EntPillorder> {
        const response = await this.createPillorderRaw(requestParameters);
        return await response.value();
    }

    /**
     * Create pillorderitem
     * Create pillorderitem
     */
    async createPillorderitemRaw(requestParameters: CreatePillorderitemRequest): Promise<runtime.ApiResponse<EntPillorderItem>> {
        if (requestParameters.pillorderitem === null || requestParameters.pillorderitem === undefined) {
            throw new runtime.RequiredError('pillorderitem','Required parameter requestParameters.pillorderitem was null or undefined when calling createPillorderitem.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        headerParameters['Content-Type'] = 'application/json';

        const response = await this.request({
            path: `/pillorderItems`,
            method: 'POST',
            headers: headerParameters,
            query: queryParameters,
            body: EntPillorderItemToJSON(requestParameters.pillorderitem),
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntPillorderItemFromJSON(jsonValue));
    }

    /**
     * Create pillorderitem
     * Create pillorderitem
     */
    async createPillorderitem(requestParameters: CreatePillorderitemRequest): Promise<EntPillorderItem> {
        const response = await this.createPillorderitemRaw(requestParameters);
        return await response.value();
    }

    /**
     * get dentist by ID
     * Get a dentist entity by ID
     */
    async getDentistRaw(requestParameters: GetDentistRequest): Promise<runtime.ApiResponse<EntDentist>> {
        if (requestParameters.id === null || requestParameters.id === undefined) {
            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling getDentist.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/dentists/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters.id))),
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntDentistFromJSON(jsonValue));
    }

    /**
     * get dentist by ID
     * Get a dentist entity by ID
     */
    async getDentist(requestParameters: GetDentistRequest): Promise<EntDentist> {
        const response = await this.getDentistRaw(requestParameters);
        return await response.value();
    }

    /**
     * get employee by ID
     * Get a Employee entity by ID
     */
    async getEmployeeRaw(requestParameters: GetEmployeeRequest): Promise<runtime.ApiResponse<EntEmployee>> {
        if (requestParameters.id === null || requestParameters.id === undefined) {
            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling getEmployee.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/employees/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters.id))),
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntEmployeeFromJSON(jsonValue));
    }

    /**
     * get employee by ID
     * Get a Employee entity by ID
     */
    async getEmployee(requestParameters: GetEmployeeRequest): Promise<EntEmployee> {
        const response = await this.getEmployeeRaw(requestParameters);
        return await response.value();
    }

    /**
     * get patient by ID
     * Get a patient entity by ID
     */
    async getPatientRaw(requestParameters: GetPatientRequest): Promise<runtime.ApiResponse<EntPatient>> {
        if (requestParameters.id === null || requestParameters.id === undefined) {
            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling getPatient.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/patients/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters.id))),
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntPatientFromJSON(jsonValue));
    }

    /**
     * get patient by ID
     * Get a patient entity by ID
     */
    async getPatient(requestParameters: GetPatientRequest): Promise<EntPatient> {
        const response = await this.getPatientRaw(requestParameters);
        return await response.value();
    }

    /**
     * get pillorder by ID
     * Get a Pillorder entity by ID
     */
    async getPillorderRaw(requestParameters: GetPillorderRequest): Promise<runtime.ApiResponse<EntPillorder>> {
        if (requestParameters.id === null || requestParameters.id === undefined) {
            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling getPillorder.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/pillorders/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters.id))),
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntPillorderFromJSON(jsonValue));
    }

    /**
     * get pillorder by ID
     * Get a Pillorder entity by ID
     */
    async getPillorder(requestParameters: GetPillorderRequest): Promise<EntPillorder> {
        const response = await this.getPillorderRaw(requestParameters);
        return await response.value();
    }

    /**
     * get PillorderItem by ID
     * Get a PillorderItem entity by ID
     */
    async getPillorderitemRaw(requestParameters: GetPillorderitemRequest): Promise<runtime.ApiResponse<EntPillorderItem>> {
        if (requestParameters.id === null || requestParameters.id === undefined) {
            throw new runtime.RequiredError('id','Required parameter requestParameters.id was null or undefined when calling getPillorderitem.');
        }

        const queryParameters: runtime.HTTPQuery = {};

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/pillorderitems/{id}`.replace(`{${"id"}}`, encodeURIComponent(String(requestParameters.id))),
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => EntPillorderItemFromJSON(jsonValue));
    }

    /**
     * get PillorderItem by ID
     * Get a PillorderItem entity by ID
     */
    async getPillorderitem(requestParameters: GetPillorderitemRequest): Promise<EntPillorderItem> {
        const response = await this.getPillorderitemRaw(requestParameters);
        return await response.value();
    }

    /**
     * list dentist entities
     * List dentist entities
     */
    async listDentistRaw(requestParameters: ListDentistRequest): Promise<runtime.ApiResponse<Array<EntDentist>>> {
        const queryParameters: runtime.HTTPQuery = {};

        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }

        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/dentists`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(EntDentistFromJSON));
    }

    /**
     * list dentist entities
     * List dentist entities
     */
    async listDentist(requestParameters: ListDentistRequest): Promise<Array<EntDentist>> {
        const response = await this.listDentistRaw(requestParameters);
        return await response.value();
    }

    /**
     * list employee entities
     * List employee entities
     */
    async listEmployeeRaw(requestParameters: ListEmployeeRequest): Promise<runtime.ApiResponse<Array<EntEmployee>>> {
        const queryParameters: runtime.HTTPQuery = {};

        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }

        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/employees`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(EntEmployeeFromJSON));
    }

    /**
     * list employee entities
     * List employee entities
     */
    async listEmployee(requestParameters: ListEmployeeRequest): Promise<Array<EntEmployee>> {
        const response = await this.listEmployeeRaw(requestParameters);
        return await response.value();
    }

    /**
     * list patient entities
     * List patient entities
     */
    async listPatientRaw(requestParameters: ListPatientRequest): Promise<runtime.ApiResponse<Array<EntPatient>>> {
        const queryParameters: runtime.HTTPQuery = {};

        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }

        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/patients`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(EntPatientFromJSON));
    }

    /**
     * list patient entities
     * List patient entities
     */
    async listPatient(requestParameters: ListPatientRequest): Promise<Array<EntPatient>> {
        const response = await this.listPatientRaw(requestParameters);
        return await response.value();
    }

    /**
     * list pillorder entities
     * List pillorder entities
     */
    async listPillorderRaw(requestParameters: ListPillorderRequest): Promise<runtime.ApiResponse<Array<EntPillorder>>> {
        const queryParameters: runtime.HTTPQuery = {};

        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }

        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/pillorders`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(EntPillorderFromJSON));
    }

    /**
     * list pillorder entities
     * List pillorder entities
     */
    async listPillorder(requestParameters: ListPillorderRequest): Promise<Array<EntPillorder>> {
        const response = await this.listPillorderRaw(requestParameters);
        return await response.value();
    }

    /**
     * list pillorderitem entities
     * List pillorderitem entities
     */
    async listPillorderitemRaw(requestParameters: ListPillorderitemRequest): Promise<runtime.ApiResponse<Array<EntPillorderItem>>> {
        const queryParameters: runtime.HTTPQuery = {};

        if (requestParameters.limit !== undefined) {
            queryParameters['limit'] = requestParameters.limit;
        }

        if (requestParameters.offset !== undefined) {
            queryParameters['offset'] = requestParameters.offset;
        }

        const headerParameters: runtime.HTTPHeaders = {};

        const response = await this.request({
            path: `/pillorderitems`,
            method: 'GET',
            headers: headerParameters,
            query: queryParameters,
        });

        return new runtime.JSONApiResponse(response, (jsonValue) => jsonValue.map(EntPillorderItemFromJSON));
    }

    /**
     * list pillorderitem entities
     * List pillorderitem entities
     */
    async listPillorderitem(requestParameters: ListPillorderitemRequest): Promise<Array<EntPillorderItem>> {
        const response = await this.listPillorderitemRaw(requestParameters);
        return await response.value();
    }

}
