// plugin-welcome/src/services/api/models/EntEmployee.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntEmployeeEdges,
    EntEmployeeEdgesFromJSON,
    EntEmployeeEdgesFromJSONTyped,
    EntEmployeeEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntEmployee
 */
export interface EntEmployee {
    /**
     * 
     * @type {EntEmployeeEdges}
     * @memberof EntEmployee
     */
    edges?: EntEmployeeEdges;
    /**
     * EmployeeEmail holds the value of the "employee_email" field.
     * @type {string}
     * @memberof EntEmployee
     */
    employeeEmail?: string;
    /**
     * EmployeeName holds the value of the "employee_name" field.
     * @type {string}
     * @memberof EntEmployee
     */
    employeeName?: string;
    /**
     * EmployeePassword holds the value of the "employee_password" field.
     * @type {string}
     * @memberof EntEmployee
     */
    employeePassword?: string;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntEmployee
     */
    id?: number;
}

export function EntEmployeeFromJSON(json: any): EntEmployee {
    return EntEmployeeFromJSONTyped(json, false);
}

export function EntEmployeeFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntEmployee {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'edges': !exists(json, 'edges') ? undefined : EntEmployeeEdgesFromJSON(json['edges']),
        'employeeEmail': !exists(json, 'employee_email') ? undefined : json['employee_email'],
        'employeeName': !exists(json, 'employee_name') ? undefined : json['employee_name'],
        'employeePassword': !exists(json, 'employee_password') ? undefined : json['employee_password'],
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntEmployeeToJSON(value?: EntEmployee | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'edges': EntEmployeeEdgesToJSON(value.edges),
        'employee_email': value.employeeEmail,
        'employee_name': value.employeeName,
        'employee_password': value.employeePassword,
        'id': value.id,
    };
}
