// plugin-welcome/src/services/api/models/EntPillorder.ts
/* tslint:disable */
/* eslint-disable */
/**
 * SUT SA Example API Playlist Vidoe
 * This is a sample server for SUT SE 2563
 *
 * The version of the OpenAPI document: 1.0
 * Contact: support@swagger.io
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { exists, mapValues } from '../runtime';
import {
    EntPillorderEdges,
    EntPillorderEdgesFromJSON,
    EntPillorderEdgesFromJSONTyped,
    EntPillorderEdgesToJSON,
} from './';

/**
 * 
 * @export
 * @interface EntPillorder
 */
export interface EntPillorder {
    /**
     * PillorderDate holds the value of the "PillorderDate" field.
     * @type {string}
     * @memberof EntPillorder
     */
    pillorderDate?: string;
    /**
     * PillorderNameID holds the value of the "PillorderNameID" field.
     * @type {string}
     * @memberof EntPillorder
     */
    pillorderNameID?: string;
    /**
     * 
     * @type {EntPillorderEdges}
     * @memberof EntPillorder
     */
    edges?: EntPillorderEdges;
    /**
     * ID of the ent.
     * @type {number}
     * @memberof EntPillorder
     */
    id?: number;
}

export function EntPillorderFromJSON(json: any): EntPillorder {
    return EntPillorderFromJSONTyped(json, false);
}

export function EntPillorderFromJSONTyped(json: any, ignoreDiscriminator: boolean): EntPillorder {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    return {
        
        'pillorderDate': !exists(json, 'PillorderDate') ? undefined : json['PillorderDate'],
        'pillorderNameID': !exists(json, 'PillorderNameID') ? undefined : json['PillorderNameID'],
        'edges': !exists(json, 'edges') ? undefined : EntPillorderEdgesFromJSON(json['edges']),
        'id': !exists(json, 'id') ? undefined : json['id'],
    };
}

export function EntPillorderToJSON(value?: EntPillorder | null): any {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        
        'PillorderDate': value.pillorderDate,
        'PillorderNameID': value.pillorderNameID,
        'edges': EntPillorderEdgesToJSON(value.edges),
        'id': value.id,
    };
}
